### PR TITLE
Post: Purpose of Django's SECRET_KEY

### PR DESCRIPTION
**Pull request by SanketDG**
Looking for comments on the rough outline @ankitsurkar @mihirsn

### CODE REVIEW

File: content/post/django-secret-key.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+date: 2020-07-24
+description: "Diving deep into the Django's to learn where SECRET_KEY is used"
+featured_image: "https://images.unsplash.com/photo-1537111493080-d1bfb2e415ad?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80"
+tags: []
+title: "What is the purpose of SECRET_KEY in Django?"
```
Comment (mihirsn) [inline content/post/django-secret-key.md]: What about title: Purpose of SECRET_KEY in Django

File: content/post/django-secret-key.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+The `SECRET_KEY` setting in Django is like Faizal from Gangs of Wasseypur part 1, we know this character exists but we do not know what his purpose is
+
+What exactly is `SECRET_KEY` used for?
+
+For most cases, it is used as a salt. But What's a salt?
+
+It's used as the salt in the password reset token of any user.
+
+It's also used as the when a user logs in and the session token is generated.
```
Comment (mihirsn) [inline content/post/django-secret-key.md]: can be converted to - It's also used when a user logs in and the session token is generated

**Review by mihirsn — CHANGES_REQUESTED** [review]

**Comment by mihirsn** [comment]
I think we can add one more point about its use : such that in lot of applications developers keep it hard coded in settings.py file instead we can set environment variable from where we can access it.

File: content/post/django-secret-key.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+The session token is generated using a HMAC
+
+It's also used to sign messages in a cookie.
+
+For signing JSON, see signing https://docs.djangoproject.com/en/3.1/topics/signing/
+
+4.0 Breaking changes
```
Comment (mihirsn) [inline content/post/django-secret-key.md:26]: ___Django___ 4.0 Breaking changes ?

Comment (mihirsn) [inline content/post/django-secret-key.md]: ___Django___ 4.0 Breaking changes ?

File: content/post/django-secret-key.md (hunk 1 above)
Comment (SanketDG) [reply content/post/django-secret-key.md]: Will change

**Pull request merged**